fidesctl/src/fidesctl/cli/utils.py

```diff
@@ -8,13 +8,30 @@
 from typing import Any, Callable, Dict
 
 import click
+import requests
 from fideslog.sdk.python.event import AnalyticsEvent
 from fideslog.sdk.python.exceptions import AnalyticsException
 from fideslog.sdk.python.utils import OPT_OUT_COPY
-from requests import Response
 
 from fidesctl.core.config.utils import get_config_from_file, update_config_file
-from fidesctl.core.utils import echo_red
+from fidesctl.core.utils import echo_red, check_response
+from fidesctl.core import api as _api
+
+
+def check_server(cli_version: str, server_url: str) -> None:
+    "Runs a health check and a version check against the server."
+    healthcheck_url = server_url + "/health"
+    try:
+        health_response = check_response(_api.ping(healthcheck_url))
+    except requests.exceptions.ConnectionError as err:
+        echo_red("Connection failed, webserver is unreachable.")
+        raise SystemExit(err)
+
+    server_version = health_response.json()["data"]["version"]
+    if str(server_version) != str(cli_version):
+        echo_red(
+            f"Mismatched versions!\nServer Version: {server_version}\nCLI Version: {cli_version}"
+        )
```

Comment:
Per my previous comment, I'd rather we raise a custom exception here and handle logging in the `status` handler.

Reply:
not going to write a custom error for now, but will make sure it errors if the versions mismatch